pinot-query-planner/src/main/java/org/apache/calcite/prepare/PinotCalciteCatalogReader.java

```diff
@@ -0,0 +1,466 @@
+  }
+
+  protected PinotCalciteCatalogReader(CalciteSchema rootSchema,
+      SqlNameMatcher nameMatcher, List<List<String>> schemaPaths,
+      RelDataTypeFactory typeFactory, CalciteConnectionConfig config) {
+    _rootSchema = Objects.requireNonNull(rootSchema, "rootSchema");
+    _nameMatcher = nameMatcher;
+    _schemaPaths =
+        Util.immutableCopy(Util.isDistinct(schemaPaths)
+            ? schemaPaths
+            : new LinkedHashSet<>(schemaPaths));
+    _typeFactory = typeFactory;
+    _config = config;
+  }
+
+  @Override public PinotCalciteCatalogReader withSchemaPath(List<String> schemaPath) {
+    return new PinotCalciteCatalogReader(_rootSchema, _nameMatcher,
+        ImmutableList.of(schemaPath, ImmutableList.of()), _typeFactory, _config);
+  }
+
+  @Override public Prepare.@Nullable PreparingTable getTable(final List<String> names) {
+    // First look in the default schema, if any.
+    // If not found, look in the root schema.
+    CalciteSchema.TableEntry entry = SqlValidatorUtil.getTableEntry(this, names);
+    if (entry != null) {
+      final Table table = entry.getTable();
+      if (table instanceof Wrapper) {
+        final Prepare.PreparingTable relOptTable =
+            ((Wrapper) table).unwrap(Prepare.PreparingTable.class);
+        if (relOptTable != null) {
+          return relOptTable;
+        }
+      }
+      return RelOptTableImpl.create(this,
+          table.getRowType(_typeFactory), entry, null);
+    }
+    return null;
+  }
+
+  @Override public CalciteConnectionConfig getConfig() {
+    return _config;
+  }
+
+  private Collection<org.apache.calcite.schema.Function> getFunctionsFrom(
+      List<String> names) {
+    final List<org.apache.calcite.schema.Function> functions2 =
+        new ArrayList<>();
+    final List<List<String>> schemaNameList = new ArrayList<>();
+    if (names.size() > 1) {
+      // Name qualified: ignore path. But we do look in "/catalog" and "/",
+      // the last 2 items in the path.
+      if (_schemaPaths.size() > 1) {
+        schemaNameList.addAll(Util.skip(_schemaPaths));
+      } else {
+        schemaNameList.addAll(_schemaPaths);
+      }
+    } else {
+      for (List<String> schemaPath : _schemaPaths) {
+        CalciteSchema schema =
+            SqlValidatorUtil.getSchema(_rootSchema, schemaPath, _nameMatcher);
+        if (schema != null) {
+          schemaNameList.addAll(schema.getPath());
+        }
+      }
+    }
+    for (List<String> schemaNames : schemaNameList) {
+      CalciteSchema schema =
+          SqlValidatorUtil.getSchema(_rootSchema,
+              Iterables.concat(schemaNames, Util.skipLast(names)), _nameMatcher);
+      if (schema != null) {
+        final String name = Util.last(names);
+        // ====================================================================
+        // LINE CHANGED BELOW
+        // ====================================================================
+        functions2.addAll(schema.getFunctions(name, false));
```

Comment:
maybe it is easier to read if written into:
boolean caseSensitive = false;
functions2.addAll(schema.getFunctions(name, caseSensitive));

Reply:
yeah i tried but checkstyle throw warning on constant usage.

Reply:
I think the "proper" way to do that would be to create:
```
private static final boolean NOT_CASE_SENSITIVE = false;
...
functions2.addAll(schema.getFunctions(name, NOT_CASE_SENSITIVE));
```